qa/os/src/test/java/org/elasticsearch/packaging/test/PackageUpgradeTests.java

```diff
@@ -106,11 +106,14 @@ public void test20InstallUpgradedVersion() throws Exception {
         assertInstalled(distribution);
         verifyPackageInstallation(installation, distribution, sh);
         verifySecurityNotAutoConfigured(installation);
+
+        // Upgrade overwrites the configuration file because we run with `--force-confnew`, so we need to remove security configuration
+        // again
+        possiblyRemoveSecurityConfiguration(installation);
```

Comment:
I know it looks suspicious that we `verifySecurityNotAutoConfigured` and then `possiblyRemoveSecurityConfiguration`; however, `verifySecurityNotAutoConfigured` does not actually check if security is enabled or not.

Reply:
Here's my understanding of the issue:
The BWC distrib is installed without Security, either because it is older than 8.0 or, if it is newer, the auto-configuration is reverted after installation and before starting the node.
When going from such a BWC install, the upgraded distribution is also NOT going to be auto-configured. We assert that, and that's good.
But even if the new distrib is not auto-configured for Security, Security (and authentication) still defaults to enabled.

Based on the above understanding, I suggest we use this alternative approach, which I believe is cleaner in its intention to disable security (but not refer to auto-configuration):

```
diff --git a/qa/os/src/test/java/org/elasticsearch/packaging/test/PackageUpgradeTests.java b/qa/os/src/test/java/org/elasticsearch/packaging/test/PackageUpgradeTests.java
index f4a4fbf79ca..5ebee7d80ec 100644
--- a/qa/os/src/test/java/org/elasticsearch/packaging/test/PackageUpgradeTests.java
+++ b/qa/os/src/test/java/org/elasticsearch/packaging/test/PackageUpgradeTests.java
@@ -97,15 +97,14 @@ public class PackageUpgradeTests extends PackagingTestCase {
         } else {
             installation = Packages.upgradePackage(sh, distribution);
         }
-        // We add this so that we don't trigger the SecurityImplicitBehaviorBootstrapCheck in 8
-        if (Version.fromString(bwcDistribution.baseVersion).before(Version.V_8_0_0)
-            && Version.fromString(distribution.baseVersion).onOrAfter(Version.V_8_0_0)) {
-            ServerUtils.addSettingToExistingConfiguration(installation, "xpack.security.enabled", "false");
-        }

         assertInstalled(distribution);
         verifyPackageInstallation(installation, distribution, sh);
         verifySecurityNotAutoConfigured(installation);
+
+        // TODO modify tests to work with Security
+        // Security is still enabled by default even if not autoconfigured
+        ServerUtils.addSettingToExistingConfiguration(installation, "xpack.security.enabled", "false");
     }

     @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/85386")
```

Reply:
I put the `addSettingToExistingConfiguration` at the end so that `assertInstalled`, `verifyPackageInstallation`, `verifySecurityNotAutoConfigured` get to run and do their stuff on the genuine files set by the upgrade process.
The bootstrap checks only run when the node starts, which is in the following test, so maybe we should move `addSettingToExistingConfiguration` there, but I don't have strong preferences.

Reply:
> The bootstrap checks only run when the node starts, which is in the following test, so maybe we should move addSettingToExistingConfiguration there, but I don't have strong preferences.

I think it's all good at the end of the current test.